src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -50,15 +50,15 @@ public NetworkStream(Socket socket, FileAccess access, bool ownsSocket)
                 // allowing non-blocking sockets could result in non-deterministic failures from those
                 // operations. A developer that requires using NetworkStream with a non-blocking socket can
                 // temporarily flip Socket.Blocking as a workaround.
-                throw GetCustomException(SR.net_sockets_blocking);
+                throw new IOException(SR.net_sockets_blocking);
```

Comment:
I originally added the GetCustomException stuff as part of the NetworkException work. It might be worth keeping in case we decide to introduce that back again.

Reply:
That said, it looks like it's only used a couple places now, so it probably isn't worth it...

Reply:
I think it's not worth to use helper methods in the constructor.

On other places the new helper `WrapException` is being used. If we decide to reintroduce `NetworkException` later we can do the trick to extract `SocketError` right in that method: `if (exception is SocketException socketException)`